js/web/lib/wasm/jsep/webgpu/ops/gelu.ts

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {DataType} from '../../../wasm-common';
+import {TensorView} from '../../tensor';
+import {ShapeUtil} from '../../util';
+import {ComputeContext, GpuDataType, ProgramInfo, ProgramInfoLoader, ProgramMetadata} from '../types';
+
+import {ShaderHelper} from './common';
+import {erfImpl} from './unary-op';
+
+const validateInputs = (inputs: readonly TensorView[]): void => {
+  if (!inputs || inputs.length !== 1) {
+    throw new Error('Gelu requires 1 input');
+  }
+  if (inputs[0].dataType !== DataType.float) {
+    throw new Error('Input must be float');
+  }
+};
+
+
+const createGeluProgramInfo = (metadata: ProgramMetadata, inputs: readonly TensorView[]): ProgramInfo => {
+  const inputShape = inputs[0].dims;
+  const outputShape = inputShape.slice(0);
+  const outputSize = ShapeUtil.size(outputShape);
+  const dataType = 'f32';
+  const getShaderSource = (shaderHelper: ShaderHelper) => `
+    ${erfImpl('f32')};
```

Comment:
Is Gelu able to be implemented as vec4<f32>?

Reply:
Gelu is an element-wise operation. Hence it is better to implement it as such. The input may not be divisible by 4, we need extra logic to handle boundary condition.

Reply:
Is there any problem with factoring out the ERF function?